lexicon/providers/mythicbeasts.py

```diff
@@ -0,0 +1,280 @@
+"""Module provider for Mythic Beasts"""
+from __future__ import absolute_import
+
+import hashlib
+import json
+import logging
+import time as time
+
+import requests
+
+from lexicon.providers.base import Provider as BaseProvider
+
+LOGGER = logging.getLogger(__name__)
+
+NAMESERVER_DOMAINS = ["mythic-beasts.com"]
+
+
+def provider_parser(subparser):
+    """Return the parser for this provider"""
+    subparser.description = """
+        There are two ways to provide an authentication granting access to the Mythic Beasts API
+        1 - With your API credentials (user/password),
+            with --auth-username and --auth-password flags.
+        2 - With an API token, using --auth-token flags.
+        These credentials and tokens must be generated using the Mythic Beasts API v2.
+    """
+    subparser.add_argument(
+        "--auth-username",
+        help="specify API credentials username",
+    )
+    subparser.add_argument(
+        "--auth-password",
+        help="specify API credentials password",
+    )
+    subparser.add_argument(
+        "--auth-token",
+        help="specify API token for authentication",
+    )
+
+
+class Provider(BaseProvider):
+    """Provider class for Mythic Beasts"""
+
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
+        self.domain_id = None
+        self.api_endpoint = "https://api.mythic-beasts.com/dns/v2"
+        self.auth_token = None
+
+    def _authenticate(self):
+        payload = self._get("/zones")
+
+        if self.domain is None:
```

Comment:
In fact by construction, `self.domain` is always set by construction in the parent class (see https://github.com/AnalogJ/lexicon/blob/59419ffba7797a860251c83f73221054b3f72862/lexicon/providers/base.py#L62).

The intention in `_authenticate()` is doing the following things:
1) retrieve the credentials required to authenticate from the provider options
2) authenticate against the API and retrieve the tokens that will be used for the lifetime of the provider instance for all HTTP calls
3) verify that with these tokens we are authorized to manipulate the DNS zone targeted by `self.domain`
4) set the internal domain id on `self.domain_id` for the lifetime of the provider instance

So basically I think that here you should make the autentication token retrieval (if not set by the caller), then ensure that you can see the target zone specified by `self.domain`, then assign its value to `self.domain_id` (you do not have necessarily to hash it, it can be the domain itself).